src/main/java/org/folio/services/finance/transaction/BaseTransactionService.java

```diff
@@ -74,63 +71,57 @@ private Future<TransactionCollection> getTransactionsChunk(List<String> transact
     return this.getTransactions(query, 0, transactionIds.size(), requestContext);
   }
 
-  public Future<Transaction> createTransaction(Transaction transaction, RequestContext requestContext) {
-    return Optional.ofNullable(getEndpoint(transaction))
-        .map(RequestEntry::new)
-        .map(requestEntry -> restClient.post(requestEntry, transaction, Transaction.class, requestContext))
-        .orElseGet(this::unsupportedOperation);
+  public Future<Void> batchAllOrNothing(List<Transaction> transactionsToCreate, List<Transaction> transactionsToUpdate,
+      List<String> idsOfTransactionsToDelete, List<TransactionPatch> transactionPatches, RequestContext requestContext) {
+    Batch batch = new Batch();
+    if (transactionsToCreate != null) {
+      transactionsToCreate.forEach(tr -> {
+        if (tr.getId() == null) {
```

Comment:
StringUtils is better to avoid empty strings

Reply:
Regarding that, I think there is already a type validation in the raml for the ids (UUID). I feel like checking if the id is an empty string is redundant, and an id with an empty string should cause a validation error rather than a silent replacement. Also I prefer to avoid using external libraries if possible.